vehicle-simulator/main.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2023 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/signal"
+	"syscall"
+
+	dittoModel "github.com/eclipse/ditto-clients-golang/model"
+	"github.com/eclipse/ditto-clients-golang/protocol"
+	"github.com/eclipse/ditto-clients-golang/protocol/things"
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+)
+
+type DeviceData struct {
+	DeviceID string `json:"deviceID"`
+	TenantID string `json:"tenantID"`
+	PolicyID string `json:"policyID"`
+}
+
+var mqttClient mqtt.Client
+var deviceData = DeviceData{}
+var telemetrySenderStarted bool
+var selectedConnectorDeviceID string // holds the last deviceID from the previous active connector
```

Comment:
_selectedConnectorDeviceID_ seems redundant as it is kept within the _deviceData_ variable as well.